src/ray/ray_impl.rs

```diff
@@ -106,6 +110,48 @@ impl<T: Scalar + Copy, const D: usize> Ray<T, D> {
         self.ray_intersects_aabb(aabb)
     }
 
+    /// Intersect [`Aabb`] by [`Ray`]
+    /// Returns slice of intersections, two numbers `T`
+    /// where the first number is the distance from [`Ray`] to the nearest intersection point
+    /// and the second number is the distance from [`Ray`] to the farthest intersection point
+    ///
+    /// If there are no intersections, it returns negative one for both distances
+    pub fn intersection_slice_for_aabb(&self, aabb: &Aabb<T, D>) -> (T, T)
```

Comment:
This function needs some unit tests (I know there are some already for the distance traversal, but this one needs its own tests.

Reply:
@dashedman as far as I can see, we don't have any unit tests for this function here still.